Expensify/App
# Reintroduce workspace new without a modal, just the url

CC: @iwiznia, @cead22
### Details
Reintroduces the Workspace/New command which will be used in inboxTasks and the pricing page from OldDot.

Note: this PR includes changes from this one and is held on it:
- [x] https://github.com/Expensify/App/pull/5387

### Fixed Issues
related to Expensify/Expensify#177739

### Tests
Same as QA done locally.

Make sure you have these other PRs running locally
Expensify/Web-Expensify#31977
Expensify/Auth#5963



### QA Steps
Flow 1: Unvalidated new accounts
1. Create a new account in OldDot that has access to the freePlan beta.
2. Go to settings -> policy -> group and select the free plan here:
![image](https://user-images.githubusercontent.com/19364431/134283153-e83d0550-eff5-47f7-9b05-e21fcaa364e6.png)

3. Verify that a new tab opens and you are logged into NewDot and you are seeing the Workspace Settings page for a newly created Workspace:
![image](https://user-images.githubusercontent.com/19364431/134283736-52c37020-a9f4-4f6d-a91c-b8eba0024e8b.png)
4. Wait a minute
5. Do any action (like add a comment)
6. Check you are not logged out

Flow 2: Existing accounts
1. Log into an account on NewDot Expensify, create a Workspace if you don't already have one using the `+` on the bottom right.
2. Log out of NewDot and then log into that same account on OldDot and go to Settings -> Policy and find your Workspace there. It should have a card icon like these:
![image](https://user-images.githubusercontent.com/19364431/134262812-35531673-c2f0-48ab-9dbb-f58cced40371.png)
3. Click on its name and verify it opens a new tab, automatically logs you in, and brings you to the workspace settings page:
![image](https://user-images.githubusercontent.com/19364431/134262925-c4fd8487-f084-4ffc-98fb-2ef25bfa6c3d.png)
4. Wait a minute
5. Do any action (like add a comment)
6. Check you are not logged out

### Tested On


### Screenshots
Included in QA
